Mathlib/Analysis/NormedSpace/MultipliableUniformlyOn.lean

```diff
@@ -0,0 +1,130 @@
+  have : ∀ᶠ i in p, ∀ x ∈ K, (f i x).re ≤ v + 1 := hf.re.eventually_forall_le (lt_add_one v) hv
+  refine (UniformContinuousOn.cexp _).comp_tendstoUniformlyOn_eventually (by simpa) ?_ hf
+  simpa using fun a ha ↦ (hv a ha).trans (lt_add_one v).le
+
+lemma Complex.hasSumUniformlyOn_log_one_add {α : Type*} {f : ι → α → ℂ} (K : Set α)
+    {u : ι → ℝ} (hu : Summable u) (h : ∀ᶠ n in cofinite, ∀ x ∈ K, ‖f n x‖ ≤ u n) :
+    HasSumUniformlyOn (fun i x ↦ log (1 + f i x)) (fun x ↦ ∑' i, log (1 + f i x)) {K} := by
+  simp only [hasSumUniformlyOn_iff_tendstoUniformlyOn, Set.mem_singleton_iff, forall_eq]
+  apply tendstoUniformlyOn_tsum_of_cofinite_eventually <| hu.mul_left (3 / 2)
+  filter_upwards [h, hu.tendsto_cofinite_zero.eventually_le_const one_half_pos] with i hi hn' x hx
+    using (norm_log_one_add_half_le_self <| (hi x hx).trans hn').trans (by simpa using hi x hx)
+
+lemma Complex.tendstoUniformlyOn_tsum_nat_log_one_add {α : Type*} {f : ℕ → α → ℂ} (K : Set α)
+    {u : ℕ → ℝ} (hu : Summable u) (h : ∀ᶠ n in atTop, ∀ x ∈ K, ‖f n x‖ ≤ u n) :
+    TendstoUniformlyOn (fun n x ↦ ∑ i ∈ Finset.range n, Complex.log (1 + f i x))
+    (fun x ↦ ∑' i : ℕ, Complex.log (1 + f i x)) atTop K := by
+  rw [← Nat.cofinite_eq_atTop] at h
+  simpa only [Set.mem_singleton_iff, forall_eq] using
+    hasSumUniformlyOn_tendstoUniformlyOn_nat (Complex.hasSumUniformlyOn_log_one_add K hu h)
+
+end cexp_clog
+
+section UniformlyOn
+
+lemma hasProdUniformlyOn_of_clog {f : ι → α → ℂ} {𝔖 : Set (Set α)}
+    (hf : SummableUniformlyOn (fun i x ↦ log (f i x)) 𝔖) (hfn : ∀ K ∈ 𝔖, ∀ x ∈ K, ∀ i, f i x ≠ 0)
+    (hg : ∀ K ∈ 𝔖, BddAbove <| (fun x ↦ (∑' n : ι, log (f n x)).re) '' K) :
+    HasProdUniformlyOn f (fun x ↦ ∏' i, f i x) 𝔖 := by
+  simp only [hasProdUniformlyOn_iff_tendstoUniformlyOn, Set.mem_singleton_iff, forall_eq]
+  obtain ⟨r, hr⟩ := hf.exists
+  intro K hK
+  suffices H : TendstoUniformlyOn (fun s a ↦ ∏ i ∈ s, f i a) (cexp ∘ r) atTop K by
+    apply TendstoUniformlyOn.congr_right H
+    apply Set.EqOn.trans (Set.EqOn.comp_left (hr.tsum_eqOn hK)).symm
+    exact fun x hx ↦ (cexp_tsum_eq_tprod (hfn K hK x hx) (hf.summable hK hx))
+  have h1 := hr.tsum_eqOn hK
+  simp only [hasSumUniformlyOn_iff_tendstoUniformlyOn, Set.mem_singleton_iff, forall_eq] at hr
+  apply TendstoUniformlyOn.congr ((hr K hK).comp_cexp ?_)
+  · filter_upwards with s i hi using by simp [exp_sum, fun y ↦ exp_log (hfn K hK i hi y)]
+  · convert hg K hK
+    simp_all only [h1 _, ne_eq]
+
+lemma multipliableUniformlyOn_of_clog {f : ι → α → ℂ} {𝔖 : Set (Set α)}
+    (hf : SummableUniformlyOn (fun i x ↦ log (f i x)) 𝔖) (hfn : ∀ K ∈ 𝔖, ∀ x ∈ K, ∀ i, f i x ≠ 0)
+    (hg : ∀ K ∈ 𝔖, BddAbove <| (fun x ↦ (∑' n : ι, log (f n x)).re) '' K) :
+    MultipliableUniformlyOn f 𝔖 :=
+    ⟨fun x ↦ ∏' i, f i x, hasProdUniformlyOn_of_clog  hf hfn hg⟩
+
+lemma hasProdUniformlyOn_nat_one_add [TopologicalSpace α] {f : ℕ → α → ℂ} {K : Set α}
```

Comment:
Can we have a version of this with a general index set instead of Nat? It should also probably be in the `Summable` namespace. Ditto for the remaining 3 lemmas below.